spock-core/src/main/java/org/spockframework/mock/runtime/JavaMockInterceptor.java

```diff
@@ -62,6 +62,17 @@ public Object intercept(Object target, Method method, Object[] arguments, IRespo
           return GroovyRuntimeUtil.invokeMethod(target, methodName, GroovyRuntimeUtil.asArgumentArray(normalizedArgs[1]));
         }
       }
+      if (isMethod(method, "getProperty", String.class)) {
+        //Groovy 3 started to call go.getProperty("x") method instead of go.getX() directly for go.x
+        Throwable throwable = new Throwable();
+        StackTraceElement mockCaller = throwable.getStackTrace()[3];
+        if (!("groovy.lang.GroovyObject$getProperty".equals(mockCaller.getClassName()) && "call".equals(mockCaller.getMethodName()))) {
```

Comment:
Please include a `//HACK:` regarding scanning the stacktrace, and maybe extend the if with a `isGroovy3()` check

Reply:
There is a similar - already existing - "hack" with a stacktrace a few lines above for `setProperty()`. I should mark it as well in that case.

Reply:
I overslept that case and - even it seems to work also for Groovy 2 - I agree it would be good to included `isGroovy3()` runtime check for that "hack". It will minimize a chance for regression for Groovy 2 and will allow us to release 2.0-M2 with more confidence.

Reply:
Hmm, it turned out that the following fails with Groovy 2 and with 1.3-groovy-2.5:
```
    def "can stub property access"() {
      Person person = Stub(Person)
      person.getName() >> "fred"

      expect:
      person.name == "fred"
      person.getProperty("name") == "fred"  //fails with 1.3-groovy-2.5 and 2.0-M1-groovy-2.5 
    }
```
It doesn't seem to be often used if wasn't reported, so I may left it as it for Groovy 2.5.

However, for Groovy 3 my change is needed to have even:
```
      person.name == "fred"
```
passing. And along the way it fixes `person.getProperty("name") == "fred"`.

Concluding, I keep the changes only for Groovy 3, waiting for some better idea to determine that in on the Groovy side (before 2.0-final is released).